core/plugins/org.fusesource.ide.foundation.core/src/org/fusesource/ide/foundation/core/util/CamelUtils.java

```diff
@@ -183,4 +191,58 @@ public boolean visit(IResource resource) throws CoreException {
 		}
 		return files;
 	}
+	
+	/**
+	 * retrieves the camel design editor
+	 * 
+	 * @return
+	 */
+	private static IEditorPart getDiagramEditor() {
```

Comment:
yes private but duplicated code.
Maybe we can extract it to a common plugin for both. I can take a look in a moment.